apps/test/integration/util/runLevelTest.js

```diff
@@ -53,28 +53,32 @@ module.exports = function (testCollection, testData, dataItem, done) {
 
   // Validate successful solution.
   var validateResult = function (report) {
-    assert(testData.expected, 'Have expectations');
-    assert(Object.keys(testData.expected).length > 0, 'No expected keys specified');
-    Object.keys(testData.expected).forEach(function (key) {
-      if (report[key] !== testData.expected[key]) {
-        var failureMsg = 'Failure for key: ' + key + '. Expected: ' + testData.expected[key] + '. Got: ' + report[key] + '\n';
-        assert(false, failureMsg);
-      }
-    });
+    try {
```

Comment:
I've wrapped this section of `validateResult` in a try/catch that calls the `done` callback with any thrown error, because it lets us get useful assertion failures when an async level test fails its custom validator.

Reply:
👍